app/code/Magento/CatalogInventory/Api/Data/WarehouseRecordInterface.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace Magento\CatalogInventory\Api\Data;
+use Magento\Framework\Api\ExtensibleDataInterface;
+
+/**
+ * Warehouse record interface
+ *
+ * It is used only to manage stock status
+ * qty of non virtual warehouse inventory
+ *
+ * @api
+ */
+interface WarehouseRecordInterface extends ExtensibleDataInterface
+{
+    /**
+     * Returns identifier of inventory record
+     *
+     * @return int
+     */
+    public function getId();
+
+    /**
+     * Returns identifier of assigned product
+     *
+     * @return int
+     */
+    public function getProductId();
+
+    /**
+     * Returns identifier of assigned inventory
+     *
+     * @return int
+     */
+    public function getInventoryId();
+
+    /**
+     * Returns available quantity of product in the specified warehouse
+     *
+     * @return float
+     */
+    public function getQuantity();
+
+    /**
+     * Sets available quantity of product in the specified warehouse
+     *
+     * @param float $quantity
+     * @return $this
+     */
+    public function setQuantity($quantity);
+
+    /**
+     * Sets a stock status for an item
+     *
+     * @param int $flag
+     * @return $this
+     */
+    public function setIsSalable($flag);
+
+    /**
+     * Sets availability status code for warehouse record
+     *
+     * @param string $code
+     * @return $this
+     */
+    public function setAvailabilityStatusCode($code);
+
+    /**
+     * Returns availability status code for warehouse record
+     *
+     * @return string
+     */
+    public function getAvailabilityStatusCode();
+
+    /**
+     * Returns instance of inventory interface
+     *
+     * Object should be set by inventory record repository
+     *
+     * @return InventoryInterface
+     */
+    public function getInventory();
```

Comment:
{quote}
Maybe it makes sense to create something like the following:

InventoryInterface -> InventoryDataInterface 
WarehouseInterface -> InventoryDataInterface

In this case, InventoryDataInterface contains all getter methods, that are shared between warehouse entity and inventory entity.
{quote}
Yes, this looks reasonable for me.

And seems we need to do the same for interfaces:
WarehouseRecordInterface
InventoryRecordInterface
because these interfaces looks identical as well